frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/providers/ovirt/ovirt-state-update.ts

```diff
@@ -61,6 +62,19 @@ const startControllerAndCleanup = (options: UpdateOptions) => {
   cleanupOvirtProvider(options); // call should be idempotent and called on every provider change
 
```

Comment:
this whole function `startControllerAndCleanup` will get called on every provider change

so we have to update the validations unconditionally every time like this:

```ts
dispatch(
  vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](id, VMSettingsField.NAME, {
    validations:  isOvirtProvider(state, id) ? [
      {
        type: UIValidationType.LENGTH,
        settings: {
          min: 1,
          max: 63,
        },
      },
    ] : [],
  }),
);
```

Reply:
I made this change, but it doesn't look like the `validations` property is changing back to `[]` when the provider changes away from ovirt.

Reply:
because you have it under `isOvirtProvider(state, id)` It should be called unconditionally (except the initil provider change condition)

Reply:
I took it outside of the conditional block, but it still isn't changing. I verified that this `dispatch` call has the right value, but haven't figure out why the reducer isn't doing its job, yet.

Reply:
because you are calling `UpdateVmSettingsField` action which has a special implementation for merging fields in `mergeDeepInSpecial` . So it wants to update the field and not set it.

Can you try to use `InternalActionType.SetInVmSettings` action for just setting the value?

note: also if we once add such validations to vmware provider, we should not reset the values here. Not ideal solution at the moment. Maybe we could add a comment describing this behaviour?

Reply:
Fixed!